server/repository/src/db_diesel/encounter.rs

```diff
@@ -152,6 +169,18 @@ fn create_filtered_query<'a>(filter: Option<EncounterFilter>) -> BoxedProgramQue
             apply_string_filter!(sub_query, document_data, latest_document::data);
             query = query.filter(encounter_dsl::document_name.eq_any(sub_query));
         }
+
+        if patient.is_some() {
+            let patient_ids =
+                PatientRepository::create_filtered_query(patient, None).select(name_dsl::id);
+            query = query.filter(encounter_dsl::patient_id.eq_any(patient_ids));
+        }
+
+        if program.is_some() {
+            let program_ids = ProgramEnrolmentRepository::create_filtered_query(program)
```

Comment:
Program filters aren't implemented on graphql layer, so just though to reuse program enrolment...

Reply:
nice